src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs

```diff
@@ -117,6 +117,26 @@ public abstract class RestoreBackupTaskBase
 
             _restoringToDefaultDataDirectory = IsDefaultDataDirectory(RestoreFromConfiguration.DataDirectory, RestoreFromConfiguration.DatabaseName);
         }
+        
+        protected async Task<Stream> CopyRemoteStreamLocally(Stream stream)
+        {
+            return await CopyRemoteStreamLocally(stream, _serverStore.Configuration.Storage.TempPath);
+        }
+
+        public static async Task<Stream> CopyRemoteStreamLocally(Stream stream, PathSetting tempPath)
+        {
+            if (stream.CanSeek)
+                return stream;
+            // This is meant to be used by ZipArchive, which will copy the data locally because is *must* be seekable.
+            // To avoid reading everything to memory, we copy to a local file instead. Note that this also ensure that we
+            // can process files > 2GB in size. https://github.com/dotnet/runtime/issues/59027
+            var tmpFolder = tempPath?.FullPath ?? Path.GetTempPath();
+            var file = new FileStream(Path.Combine(tmpFolder, Guid.NewGuid().ToString() + ".restore-local-file"), FileMode.Create, FileAccess.ReadWrite, FileShare.Read,
```

Comment:
IIRC DeleteOnClose in not working on Posix. We do have PosixFile.DeleteOnClose. Maybe it would be best to create a DeleteOnCloseFileStream class and handle that in Dispose automatically?